### PR TITLE
Congyu/api/all communication through httpservice

### PR DESCRIPTION
**Pull request by Apollo1840**

### CODE REVIEW

**Comment by Apollo1840** [comment]
@changdiqing  this RP has the highest priority, it is related to your  issue

**Comment by changdiqing** [comment]
Can be merged after resolving merge conflicts. Indents must be fixed <--- I will do this in a separate PR

**Review by changdiqing — APPROVED** [review]

**Comment by Apollo1840** [comment]
This is should be dropped

**Pull request closed without merging**
